src/openrct2/world/Climate.cpp

```diff
@@ -181,7 +181,7 @@ extern "C"
         else if (gClimateCurrentWeatherEffect == 2)
         {
             // Create new thunder and lightning
-            uint32 randomNumber = util_rand();
+            uint32 randomNumber = scenario_rand();
```

Comment:
There is an option to disable thunders. This may cause a client UI layer not to call this function and it would desync this client, that's why it is client side.